src/integration/java/io/pinecone/integration/dataPlane/ListEndpointTest.java

```diff
@@ -0,0 +1,78 @@
+    private static Index indexConnection;
+    private static AsyncIndex asyncIndexConnection;
+
+    @BeforeAll
+    public static void setUp() throws InterruptedException {
+        indexName = indexManager.getServerlessIndexName();
+        indexConnection = indexManager.getServerlessIndexConnection();
+        asyncIndexConnection = indexManager.getServerlessAsyncIndexConnection();
+    }
+
+    @AfterAll
+    public static void cleanUp() {
+        String apiKey = System.getenv("PINECONE_API_KEY");
+        Pinecone pinecone = new Pinecone.Builder(apiKey).build();
+        pinecone.deleteIndex(indexName);
+        indexConnection.close();
+        asyncIndexConnection.close();
+    }
+
+    @Test
+    public void testListEndpoint() throws InterruptedException {
+        Thread.sleep(10000); // wait for the index to be ready for operations
+
+        // Sync
+        String listResponse = indexConnection.list("example-namespace").toString();
+        assertTrue(listResponse.contains("id1"));
+        assertTrue(listResponse.contains("id2"));
+        assertTrue(listResponse.contains("cidddd3"));
+
+        String listResponseWithPrefix = indexConnection.list("example-namespace", "i").toString();
+        assertTrue(listResponseWithPrefix.contains("id1"));
+        assertTrue(listResponseWithPrefix.contains("id2"));
+        assertFalse(listResponseWithPrefix.contains("cidddd3")); // should not be in response
+
+        String listResponseWithLimit = indexConnection.list("example-namespace", 1).toString();
+        assertTrue(listResponseWithLimit.contains("cidddd3")); // (IDs returned in LIFO order)
+        assertFalse(listResponseWithLimit.contains("id2")); // should not be in response
+        assertFalse(listResponseWithLimit.contains("id1")); // should not be in response
+
+        //Async
+        ListenableFuture<ListResponse> futureResponse = asyncIndexConnection.list("example-namespace");
+        ListResponse asyncListResponse = Futures.getUnchecked(futureResponse);
+        assertTrue(asyncListResponse.toString().contains("id1"));
+        assertTrue(asyncListResponse.toString().contains("id2"));
+        assertTrue(asyncListResponse.toString().contains("cidddd3"));
+
+        ListenableFuture<ListResponse> futureResponseWithPrefix = asyncIndexConnection.list("example-namespace", "i");
+        ListResponse asyncListResponseWithPrefix = Futures.getUnchecked(futureResponseWithPrefix);
+        assertTrue(asyncListResponseWithPrefix.toString().contains("id1"));
+        assertTrue(asyncListResponseWithPrefix.toString().contains("id2"));
+        assertFalse(asyncListResponseWithPrefix.toString().contains("cidddd3"));
+
+        ListenableFuture<ListResponse> futureResponseWithLimit = asyncIndexConnection.list("example-namespace", 1);
+        ListResponse asyncListResponseWithLimit = Futures.getUnchecked(futureResponseWithLimit);
+        assertTrue(asyncListResponseWithLimit.toString().contains("cidddd3"));
```

Comment:
Not sure if this is the best way to test async stuff. ChatGPT also told me I could make a `Futures.addCallback()` that registers a callback to be executed upon completion of the `Future`? What's best practice here?

Reply:
I think for testing purposes we basically want to allow the future to execute rather than waiting on a callback and trying to assert that way. I needed to look up `Futures.getUnchecked()` but it seems like this is equivalent to what we're doing elsewhere where we're chaining `.get()` at the end.

Also looks like that's deprecated and what you've done is preferred, looks good to me! 
https://guava.dev/releases/19.0/api/docs/com/google/common/util/concurrent/Futures.html

<img width="1194" alt="Screenshot 2024-04-17 at 2 40 48 PM" src="https://github.com/pinecone-io/pinecone-java-client/assets/119623786/5a1688a9-66ef-41ac-b59b-48b4c1cb0930">

Reply:
Okay cool, I'll leave it as-is then :)